core/controllers/reader.py

```diff
@@ -328,6 +329,43 @@ def get(self, exploration_id):
                 synced_exp_user_data
                     .most_recently_reached_checkpoint_state_name)
 
+        elif self.user_id is not None:
+            exp_user_data = exp_fetchers.get_exploration_user_data(
+                self.user_id, exploration_id)
+
+            # If exp_user_data is None, it means the exploration is started
```

Comment:
This isn't entirely true, right -- don't you need a more specific check? There are other fields in exp_user_data besides the checkpoint stuff, and the model might exist for other reasons.

Reply:
Actually, exp_user_data is created when a logged-in user has opened an exploration. And since the first card itself is a checkpoint, we know for sure that a logged-in user has some checkpoint progress if the exp_user_data exists.

Reply:
Ah ok -- sorry, my bad, I think I reasoned incorrectly. Thanks!